gammapy/obs/data_store.py

```diff
@@ -0,0 +1,318 @@
+        """Check integrity, i.e. whether index table and files match.
+        """
+        logger.info('Checking event list files')
+        available = self.check_available_event_lists(logger)
+        if np.any(~available):
+            logger.warning('Number of missing event list files: {}'.format(np.invert(available).sum()))
+
+            # TODO: implement better, more complete integrity checks.
+
+    def make_table_of_files(self, observation_table=None, filetypes=['events']):
+        """Make list of files in the datastore directory.
+
+        Parameters
+        ----------
+        observation_table : `~gammapy.obs.ObservationTable` or None
+            Observation table (``None`` means select all observations).
+        filetypes : list of str
+            File types (TODO: document in a central location and reference from here).
+
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table summarising info about files.
+        """
+        if observation_table is None:
+            observation_table = ObservationTable(self.index_table)
+
+        data = []
+        for observation in observation_table:
+            for filetype in filetypes:
+                row = dict()
+                row['OBS_ID'] = observation['OBS_ID']
+                row['filetype'] = filetype
+                filename = self.filename(observation['OBS_ID'], filetype=filetype, abspath=True)
+                row['filename'] = filename
+                data.append(row)
+
+        return Table(data=data, names=['OBS_ID', 'filetype', 'filename'])
+
+    def check_available_event_lists(self, logger=None):
+        """Check if all event lists are available.
+
+        TODO: extend this function, or combine e.g. with ``make_table_of_files``.
+
+        Returns
+        -------
+        file_available : `~numpy.ndarray`
+            Boolean mask which files are available.
+        """
+        observation_table = self.index_table
+        file_available = np.ones(len(observation_table), dtype='bool')
+        for ii in range(len(observation_table)):
+            obs_id = observation_table['OBS_ID'][ii]
+            filename = self.filename(obs_id)
+            if not Path(filename).is_file():
+                file_available[ii] = False
+                if logger:
+                    logger.warning('For OBS_ID = {:06d} the event list file is missing: {}'
+                                   ''.format(obs_id, filename))
+
+        return file_available
+
+
+def _find_file(filename, dir):
+    """Find a file at an absolute or relative location.
+
+    - First tries Path(filename)
+    - Second tris Path(dir) / filename
+    - Raises FileNotFoundError if both don't exist.
+    """
+    path1 = Path(filename)
+    path2 = Path(dir) / filename
+    if path1.is_file():
+        filename = path1
+    elif path2.is_file():
+        filename = path2
+    else:
+        raise FileNotFoundError('File not found at {} or {}'.format(path1, path2))
```

Comment:
PS: I'll try to get #364 to a state where travis-ci passes and it can be merged before lunch ...